RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,105 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Harry, Andrew. 
 //  Copyright © yagom academy. All rights reserved.
 // 
 
 import Foundation
 
-print("Hello, World!")
+func printMenu() {
+    print("가위(1), 바위(2), 보(3)! <종료: 0>", terminator: ": ")
+}
 
+func readUserInput() -> String {
+    guard let userInput = readLine() else {
+        return ""
+    }
+    
+    return userInput
+}
+
+func validationUserInput(userInput: String) throws -> Int {
+    guard userInput != "" else {
+        throw InputError.invalidInput
+    }
+    guard let number = Int(userInput) else {
+        throw InputError.invalidInput
+    }
+    guard number >= 0, number <= 3 else {
+        throw InputError.invalidInput
+    }
+
+    return number
+}
+
+func generateComputerHand() -> HandShape? {
+    guard let computerHand = HandShape.allCases.randomElement() else {
+        return nil
+    }
+    
+    return computerHand
+}
+
+func generateUserHand(validationResult: Int) -> HandShape? {
+    let handShapeList: [HandShape?] = HandShape.allCases
+    
+    guard let userHand = handShapeList[validationResult - 1]  else {
+        return nil
+    }
+    
+    return userHand
+}
+
+func compareHandShape(computerHand: HandShape?, userHand: HandShape?) -> MatchResult? {
+    guard let computerHand = computerHand, let userHand = userHand else {
+        return nil
+    }
+    
+    switch (computerHand, userHand) {
+    case (HandShape.rock, HandShape.scissors),
+        (HandShape.scissors, HandShape.paper),
+        (HandShape.paper, HandShape.rock):
+        return .lose
+    case (HandShape.scissors, HandShape.rock),
+        (HandShape.paper, HandShape.scissors),
+        (HandShape.rock, HandShape.paper):
+        return .win
+    default:
+        return .draw
+    }
```

Comment:
```suggestion
    switch (computerHand, userHand) {
    case (.rock, .scissors),
        (.scissors, .paper),
        (.paper, .rock):
        return .lose
    case (.scissors, .rock),
        (.paper, .scissors),
        (.rock, .paper):
        return .win
    default:
        return .draw
    }
```

저는 보통 이렇게 구현하는 편입니다 ㅎㅎ

Reply:
넵 처음에 xcode에서 자동완성 제공이 안되어서 안되는줄 알았는데,

다시 수정해보니 잘 되네요 ! 훨씬 보기가 편한거 같습니다 👍🏼